src/credentials/CHIPOperationalCredentials.h

```diff
@@ -45,6 +46,20 @@ struct NodeCredential
     uint16_t mLen         = 0;
 };
 
+struct OperationalCredentialSerializable
```

Comment:
Is this struct is how the data will be stored in the memory or it will be compressed? I am asking because this is not the most optimal representation.

Reply:
Currently it's not compressed. What is your suggestion for a more optimal solution?

Reply:
One of the options to encode it in CHIP TLV form for storage:

```
op-credentials [operational-credentials:3] => STRUCTURE [tag-order]
{
    op-node-cert [1] : chip-certificate,
    op-ca-cert [2,optional] : chip-certificate,
    op-root-cert [3] : chip-certificate,
    op-node-keypair [4] : OCTET STRING [length 97],
}
```

In general, I don't think this is the right storage for the Node private key - but this is topic for a different discussion/PR...

Reply:
Understood, could be something we can address in a separate PR then.